### PR TITLE
Jsonlink

### PR DESCRIPTION
**Pull request by jzhang300**
* add new JsonLink component
* add new JsonLink component example
* add new JsonLink component styles
* add link out icon to Icon.js

### CODE REVIEW

**Comment by nfriedly** [comment]
Smart thinking. I haven't tested this yet, but I like the idea.

File: src/components/JsonLink.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        href
+      >
+        <span className="json-link--span">JSON</span> <Icon type="link-out" size="small" fill={Colors.blue_50} />
+      </a>
+    </div>
+  );
+}
+
+JsonLink.propTypes = {
+  rawjson: React.PropTypes.string.isRequired,
```
Comment (nfriedly) [inline src/components/JsonLink.js]: I would recommend renaming the prop to just `json`, and also accepting objects (with automatic JSON stratification). But, neither are blockers.

**Review by nfriedly — APPROVED** [review]

**Comment by jzhang300** [comment]
This was the JsonLink Component in the Visual Recognition demo that @kognate implemented 👍

File: src/components/JsonLink.js (hunk 1 above)
Comment (jzhang300) [reply src/components/JsonLink.js]: What is JSON stratification?

File: src/components/JsonLink.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        href
+      >
+        <span className="json-link--span">JSON</span> <Icon type="link-out" size="small" fill={Colors.blue_50} />
+      </a>
+    </div>
+  );
+}
+
+JsonLink.propTypes = {
+  json: React.PropTypes.string.isRequired,
```
Comment (nfriedly) [inline src/components/JsonLink.js]: sorry, "JSON stratification" was "JSON stringification" auto-corrected :/

I was thinking that this should accept either object or string.

File: src/components/JsonLink.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Icon } from './Icon.js';
+import Colors from './Colors.js';
+
+export default function JsonLink(props) {
+  const handleClick = (e) => {
+    e.preventDefault();
+    window.open(`data:application/json, ${encodeURIComponent(JSON.stringify(props.json, null, 1))}`, '_blank');
```
Comment (nfriedly) [inline src/components/JsonLink.js]: This appears to assume props.json is an object even though propTypes specifies a string right now.

I think this bit should work with either a string or an object.

**Review by nfriedly — APPROVED** [review]

**Comment by nfriedly** [comment]
Looks good. Now you'll want to merge/rebase off of master and cut out the extra container div & classes in the example ;)

**Comment by jzhang300** [comment]
Yeah I want to handle the merge conflict after ImagePicker2 gets merged

**Pull request merged**